scripts/releaseChangelog.js

```diff
@@ -0,0 +1,242 @@
+  // TODO: Use labels
+
+  // Filter dependency updates
+  return !commitsItem.commit.message.startsWith('Bump');
+}
+
+async function findLatestTaggedVersion(octokit) {
+  // fetch tags from the GitHub API and return the last one
+  const { data } = await octokit.request(`GET /repos/${GIT_ORGANIZATION}/${GIT_REPO}/tags`);
+  return data[0].name.trim();
+}
+
+async function main(argv) {
+  const { githubToken, lastRelease: lastReleaseInput, release } = argv;
+
+  if (!githubToken) {
+    throw new TypeError(
+      'Unable to authenticate. Make sure you either call the script with `--githubToken $token` or set `process.env.GITHUB_TOKEN`. The token needs `public_repo` permissions.',
+    );
+  }
+  // Initialize the API client
+  const octokit = new Octokit({
+    auth: githubToken,
+  });
+
+  // fetch the last tag and chose the one to use for the release
+  const latestTaggedVersion = await findLatestTaggedVersion(octokit);
+  const lastRelease = lastReleaseInput !== undefined ? lastReleaseInput : latestTaggedVersion;
+  if (lastRelease !== latestTaggedVersion) {
+    console.warn(
+      `Creating changelog for ${latestTaggedVersion}..${release} when latest tagged version is '${latestTaggedVersion}'.`,
+    );
+  }
+
+  // Now We will fetch all the commits between the chosen tag and release branch
+  /**
+   * @type {AsyncIterableIterator<Octokit.Response<Octokit.ReposCompareCommitsResponse>>}
+   */
+  const timeline = octokit.paginate.iterator(
+    octokit.repos.compareCommits.endpoint.merge({
+      owner: GIT_ORGANIZATION,
+      repo: GIT_REPO,
+      base: lastRelease,
+      head: release,
+    }),
+  );
+
+  /**
+   * @type {Octokit.ReposCompareCommitsResponseCommitsItem[]}
+   */
+  const commitsItems = [];
+  for await (const response of timeline) {
+    const { data: compareCommits } = response;
+    commitsItems.push(...compareCommits.commits.filter(filterCommit));
+  }
+
+  // Fetch all the pull Request and check if there is a section named changelog
+
+  const changeLogMessages = [];
+  await Promise.all(
+    commitsItems.map(async (commitsItem) => {
+      const searchPullRequestId = commitsItem.commit.message.match(/\(#([0-9]+)\)/);
+      if (!searchPullRequestId || !searchPullRequestId[1]) {
+        return;
+      }
+
+      const {
+        data: { body: bodyMessage },
+      } = await octokit.request('GET /repos/{owner}/{repo}/pulls/{pull_number}', {
+        owner: GIT_ORGANIZATION,
+        repo: GIT_REPO,
+        pull_number: Number(searchPullRequestId[1]),
+      });
+
+      if (!bodyMessage) {
+        return;
+      }
+
+      const changelogMotif = '# changelog';
+      const changelogIndex = bodyMessage.toLowerCase().indexOf(changelogMotif);
+      if (changelogIndex >= 0) {
+        changeLogMessages.push(
+          `From https://github.com/${GIT_ORGANIZATION}/${GIT_REPO}/pull/${
+            searchPullRequestId[1]
+          }\n${bodyMessage.slice(changelogIndex + changelogMotif.length)}`,
```

Comment:
If the PR description doesn't have "## changelog" as the last section it will include everything as the example below:

<img width="400" src="https://user-images.githubusercontent.com/42154031/150157912-375b4a53-5a9e-4ad5-9b0c-aa2d9990c197.png" alt="image" style="max-width: 100%;">

I tried to use the regex `/## changelog(.*?)(?:(##|$))/is` to capture only the part after the title and before the next title. I'm not sure if we should care about this but here it's.